engine.py

```diff
@@ -515,52 +510,112 @@ def has_ui(self):
     ##########################################################################################
     # logging
 
-    def log_debug(self, msg):
+    def _emit_log_message_FUTURE(self, handler, record):
         """
-        Log debug to the Maya script editor
+        Called by the engine to log messages in Maya script editor.
+        All log messages from the toolkit logging namespace will be passed to this method.
+
+        .. note:: This method will be called by the new logging system introduced in core v0.18.
+                  It will be enabled in January 2017 by removing the method name suffix "_FUTURE".
 
-        :param msg: The message to log
+        :param handler: Log handler that this message was dispatched from.
+        :type handler: :class:`~python.logging.LogHandler`
+        :param record: Standard python logging record.
+        :type record: :class:`~python.logging.LogRecord`
         """
-        global g_last_message_time
 
-        # get the time stamps
-        prev_time = g_last_message_time
-        current_time = time.time()
+        # Do not log debug messages when debug logging setting is off.
+        if record.levelno < logging.INFO and not self.get_setting("debug_logging", False):
+            return
+
+        # Give a standard format to the message.
+        msg = "Shotgun %s" % handler.format(record)
```

Comment:
What is the point of letting the new core logging handle the message format if the engine will try to undo it later? If core v0.18 format "[%(levelname)s %(basename)s] %(message)s" is not the right one, then it should be updated there!